File: Data Structures & Algorithms/maximum-subarray/submission-1.js
class Solution {
    /**
     * @param {number[]} nums
     * @return {number}
     */
    maxSubArray(nums) {
        let maxSum = nums[0];
        let curSum = 0;
        for(let num of nums){
            curSum = Math.max(curSum, 0)
            curSum += num;
            maxSum = Math.max(curSum, maxSum);
        }
        return maxSum;
    }
}

File: Data Structures & Algorithms/majority-element/submission-1.cpp
class Solution {
public:
    int majorityElement(vector<int>& nums) {
        std::unordered_map<int, int> freq;
        
        for(int i = 0; i < nums.size(); ++i){
            freq[nums[i]]++;
        }

        for(const auto& [key, value] : freq){
            if(value > nums.size()/2) return key;
        }
    }
};